atoti-query-analyser/src/library/inputProcessors/v1tov2.js

```diff
@@ -207,27 +218,57 @@ const parseV1 = (input, tickCallback) => {
   });
 };
 
+const parseSourceId = retrieval => {
+  const [kind, retrievalIdStr, ...tail] = retrieval.sourceId.split("#");
+  const retrievalId = Number.parseInt(retrievalIdStr);
+  if (tail.length !== 0 || !Number.isInteger(retrievalId)) {
+    throw new Error(`Bad source id: ${retrieval.sourceId}`);
+  }
+  return { kind, retrievalId };
+};
+
 const computeIfAbsent = (dic, key, value) => {
   if (!(key in dic)) {
     dic[key] = value;
   }
   return dic[key];
 };
 
-const createDependencyList = v1Structure => {
-  return Object.values(v1Structure.retrievals).reduce(
-    (acc, retrieval) => {
-      if (retrieval.parents.length === 0) {
-        computeIfAbsent(acc, "-1", []).push(retrieval.id);
-      } else {
-        retrieval.parents.forEach(rId =>
-          computeIfAbsent(acc, rId, []).push(retrieval.id)
-        );
-      }
-      return acc;
-    },
-    { "-1": [] }
-  );
+const createDependencyList = (v1Structure) => {
+  const result = {
+    dependencies: {},
+    externalDependencies: {}
+  };
+
+  Object.values(v1Structure.retrievals).forEach(retrieval => {
```

Comment:
Why switching from a `#reduce` to a `#forEach`? Know that this is the idiomatic way to handle the update of a mutable value in JavaScript.
This is also a pattern in Java, though it is more complex to use.
From what I read, I suspect that this change was driven by a personal taste.